pkg/api/types.go

```diff
@@ -1202,6 +1208,21 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string
+	// Fs represents a filesystem-based local volume
+	// +optional
+	Fs *LocalFsVolume
```

Comment:
I mainly want to avoid a situation like:  Fields A, C, and F are for file-based volumes.  Fields A, B, and D are for block-based volumes.